packages/recommend-core/src/getTrendingFacets.ts

```diff
@@ -0,0 +1,61 @@
+import { RecommendClient, TrendingFacetsQuery } from '@algolia/recommend';
+
+import { TrendingFacet } from './types';
+import { mapToRecommendations } from './utils';
+import { version } from './version';
+
+export type TrendingFacetsProps<TObject> = {
+  /**
+   * The initialized Algolia recommend client.
+   */
+  recommendClient: RecommendClient;
+  /**
+   * A function to transform the retrieved items before passing them to the component.
+   *
+   * It’s useful to add or remove items, change them, or reorder them.
+   */
+  transformItems?: (
+    items: Array<TrendingFacet<TObject>>
+  ) => Array<TrendingFacet<TObject>>;
+};
+
+export type GetTrendingFacetsResult<TObject> = {
+  recommendations: Array<TrendingFacet<TObject>>;
+};
+
+export type GetTrendingFacetsProps<TObject> = TrendingFacetsProps<TObject> &
+  TrendingFacetsQuery;
+
+export function getTrendingFacets<TObject>({
+  recommendClient,
+  transformItems = (x) => x,
+  fallbackParameters,
+  indexName,
+  maxRecommendations,
+  queryParameters,
+  threshold,
+  facetName,
+}: GetTrendingFacetsProps<TObject>) {
+  const query = {
+    fallbackParameters,
+    indexName,
+    maxRecommendations,
+    queryParameters,
+    threshold,
+    facetName,
+  };
+
+  recommendClient.addAlgoliaAgent('recommend-core', version);
+
+  return recommendClient
+    .getTrendingFacets<TObject>([query])
+    .then((response) =>
+      // Multiple identical recommended `objectID`s can be returned b
+      // the engine, so we need to remove duplicates.
```

Comment:
there's the same uniqBy comment as in the other places, but uniqBy isn't called. Is that expected?

Reply:
Yes I agree 👍